src/rust/engine/process_execution/src/nailgun/mod.rs

```diff
@@ -109,7 +107,6 @@ impl CommandRunner {
     CommandRunner {
       inner: runner,
       nailgun_pool: NailgunPool::new(),
-      async_semaphore: AsyncSemaphore::new(1),
```

Comment:
This semaphore was only guarding establishing a connection, which wasn't sufficient/useful. #12982 follows up to actually control access to spawned servers.